sdk/core/System.ClientModel/src/Options/ClientLoggingOptions.cs

```diff
@@ -0,0 +1,283 @@
+
+using System.ClientModel.Internal;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace System.ClientModel.Primitives;
+
+/// <summary>
+/// Exposes client options for logging within a <see cref="ClientPipeline"/>.
+/// </summary>
+public class ClientLoggingOptions
+{
+    private bool _frozen;
+
+    private bool? _enableLogging;
+    private bool? _enableMessageLogging;
+    private bool? _enableMessageContentLogging;
+    private int? _messageContentSizeLimit;
+    private ILoggerFactory? _loggerFactory;
+    private PipelineMessageSanitizer? _sanitizer;
+    private ChangeTrackingStringList? _allowedHeaderNames;
+    private ChangeTrackingStringList? _allowedQueryParameters;
+
+    private static HashSet<string> DefaultAllowedHeaderNames { get; } = [
+            "traceparent",
+            "Accept",
+            "Cache-Control",
+            "Connection",
+            "Content-Length",
+            "Content-Type",
+            "Date",
+            "ETag",
+            "Expires",
+            "If-Match",
+            "If-Modified-Since",
+            "If-None-Match",
+            "If-Unmodified-Since",
+            "Last-Modified",
+            "Pragma",
+            "Request-Id",
+            "Retry-After",
+            "Server",
+            "Transfer-Encoding",
+            "User-Agent",
+            "WWW-Authenticate" ];
+    private static HashSet<string> DefaultAllowedQueryParameters { get; } = ["api-version"];
+    private static PipelineMessageSanitizer DefaultSanitizer { get; } = new(DefaultAllowedQueryParameters, DefaultAllowedHeaderNames);
+
+    internal const bool DefaultEnableLogging = true;
+    internal const bool DefaultEnableMessageContentLogging = false;
+    internal const int DefaultMessageContentSizeLimit = 4 * 1024;
+    internal const double RequestTooLongSeconds = 3.0; // sec
+
+    /// <summary>
+    /// Gets or sets the implementation of <see cref="ILoggerFactory"/> to use to
+    /// create <see cref="ILogger"/> instances for logging.
+    /// </summary>
+    /// <remarks>If an ILoggerFactory is not provided, logs will be written to Event Source
+    /// instead. If an ILoggerFactory is provided, logs will be written to ILogger only and not
+    /// Event Source.</remarks>
+    /// <value>Defaults to <see cref="NullLoggerFactory"/>.</value>
+    public ILoggerFactory? LoggerFactory
+    {
+        get => _loggerFactory;
+        set
+        {
+            AssertNotFrozen();
+
+            _loggerFactory = value;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets value indicating if logging should be enabled in this client pipeline.
+    /// </summary>
+    /// <value>Defaults to <c>null</c>. If <c>null</c>, this value will be treated as <c>true</c>.</value>
+    public bool? EnableLogging
+    {
+        get => _enableLogging;
+        set
+        {
+            AssertNotFrozen();
```

Comment:
```suggestion
            AssertNotFrozen();
            
```

nit: This seems to be the only one without a space.